src/Project/ToolRunner.ts

```diff
@@ -25,29 +25,59 @@ const which = require('which');
 const K_DATA: string = 'data';
 const K_EXIT: string = 'exit';
 
+// successfully killed by calling kill() method
+export type SelfKilled = 'SelfKilled';
 
 export class ToolRunner {
   tag = this.constructor.name;  // logging tag
 
+  private child: cp.ChildProcessWithoutNullStreams|undefined = undefined;
+
+  // When the spawned process was killed by kill() method, set this true
+  private killedByMe = false;
```

Comment:
(just for fun) Suicide? :smile:

Reply:
kinda of. but more specially, parent(`ToolRunner`) kills its child. 😢